tests/unit/views/pacbio/PacbioRunIndex.spec.js

```diff
@@ -1,15 +1,41 @@
 import PacbioRunIndex from '@/views/pacbio/PacbioRunIndex'
 import Response from '@/api/Response'
 import { mount, store, Data, flushPromises, nextTick } from '@support/testHelper'
+import { createTestingPinia } from '@pinia/testing'
+import { usePacbioRunsStore } from '../../../../src/stores/pacbioRuns'
+import { vi } from 'vitest'
+
+function factory(options) {
+  const wrapperObj = mount(PacbioRunIndex, {
+    global: {
+      plugins: [createTestingPinia(options)],
+    },
+  })
+
+  const store = usePacbioRunsStore()
+  return { wrapperObj, store }
+}
 
 describe('PacbioRunIndex.vue', () => {
   let wrapper, pacbioRunIndex, mockRuns, mockVersions
 
   beforeEach(async () => {
     mockRuns = new Response(Data.PacbioRuns).deserialize.runs
-    vi.spyOn(store.state.api.traction.pacbio.runs, 'get').mockResolvedValue(Data.PacbioRuns)
-
-    wrapper = mount(PacbioRunIndex, { store })
+    const spy = vi.fn().mockResolvedValue({ success: true, data: Data.PacbioRuns.data })
+    const { wrapperObj } = factory({
```

Comment:
Again this looks quite clever but I am not entirely sure what is going on.

Reply:
The factory is a builder function, which allows to create a mock instance of pinia,  with the given options that specifies how the store mockup should behave and what state it should be in while instantiating the component. This particular example doesn't make use of the function properly because we are using the same settings for all the test cases, but it can be useful in cases where we need to mock it differently for different test cases. 
Hope that make sense.

Reply:
Sure is it worth putting this as some documentation so that when we all start using Pinia we can follow your lead?

Reply:
Sure, will do :) 👍